code/modules/mob/say.dm

```diff
@@ -109,8 +109,13 @@ var/global/list/special_channel_keys = list(
 			return L
 
 /mob/proc/is_silenced()
-	. = is_muzzled()
+	. = !!is_muzzled()
 
 /mob/proc/is_muzzled()
-	var/obj/item/mask = get_equipped_item(slot_wear_mask_str)
-	return istype(mask, /obj/item/clothing/mask/muzzle) || istype(mask, /obj/item/clothing/sealant)
+	// Can't talk with something in your mouth.
+	var/datum/inventory_slot/mouth_slot = get_inventory_slot_datum(BP_MOUTH)
+	. = mouth_slot?.get_equipped_item()
+	if(!.)
+		var/obj/item/mask = get_equipped_item(slot_wear_mask_str)
+		if(istype(mask, /obj/item/clothing/mask/muzzle) || istype(mask, /obj/item/clothing/sealant))
```

Comment:
Out of scope, but this should probably be generalized somehow...? I'm not sure 100% how that'd be done in a way that could be reused elsewhere—add vars or flags to mark clothing as tightly-fit, and to mark them as blocking air, then if both are true *and* it covers the mouth you can't speak through it? Way off in the weeds here, though.